lib/pages/timetable/model/events/uni_event.dart

```diff
@@ -1,213 +1,224 @@
+import 'package:timetable/timetable.dart';
+import '../../../../generated/l10n.dart';
+import '../../../classes/model/class.dart';
+import '../../timetable_utils.dart';
+import '../academic_calendar.dart';
+
+enum UniEventType {
+  lecture,
+  lab,
+  seminar,
+  sports,
+  semester,
+  holiday,
+  examSession,
+  other
+}
+
+extension UniEventTypeExtension on UniEventType {
+  String toLocalizedString() {
+    switch (this) {
+      case UniEventType.lecture:
+        return S.current.uniEventTypeLecture;
+      case UniEventType.lab:
+        return S.current.uniEventTypeLab;
+      case UniEventType.seminar:
+        return S.current.uniEventTypeSeminar;
+      case UniEventType.sports:
+        return S.current.uniEventTypeSports;
+      case UniEventType.semester:
+        return S.current.uniEventTypeSemester;
+      case UniEventType.holiday:
+        return S.current.uniEventTypeHoliday;
+      case UniEventType.examSession:
+        return S.current.uniEventTypeExamSession;
+      case UniEventType.other:
+        return S.current.uniEventTypeOther;
+    }
+    return S.current.uniEventTypeOther;
+  }
+
+  static List<UniEventType> get classTypes => [
+        UniEventType.lecture,
+        UniEventType.lab,
+        UniEventType.seminar,
+        UniEventType.sports
+      ];
+
+  static UniEventType fromString(String string) {
+    switch (string) {
+      case 'lab':
+        return UniEventType.lab;
+      case 'lecture':
+        return UniEventType.lecture;
+      case 'seminar':
+        return UniEventType.seminar;
+      case 'sports':
+        return UniEventType.sports;
+      case 'semester':
+        return UniEventType.semester;
+      case 'holiday':
+        return UniEventType.holiday;
+      case 'examSession':
+        return UniEventType.examSession;
+      default:
+        return UniEventType.other;
+    }
+  }
+
+  Color get color {
+    switch (this) {
+      case UniEventType.lecture:
+        return Colors.pinkAccent;
+      case UniEventType.lab:
+        return Colors.blueAccent;
+      case UniEventType.seminar:
+        return Colors.orangeAccent;
+      case UniEventType.sports:
+        return Colors.greenAccent;
+      case UniEventType.semester:
+        return Colors.transparent;
+      case UniEventType.holiday:
+        return Colors.yellow;
+      case UniEventType.examSession:
+        return Colors.red;
+      case UniEventType.other:
+        return Colors.white;
+    }
+    return Colors.white;
+  }
+}
+
+class UniEvent {
+  const UniEvent({
+    @required this.start,
+    @required this.period,
+    @required this.id,
+    this.name,
+    this.location,
+    this.color,
+    this.type,
+    this.classHeader,
+    this.calendar,
+    this.relevance,
+    this.degree,
+    this.addedBy,
+    bool editable,
+  }) : editable = editable ?? true;
+
+  final String id;
+  final Color color;
+  final UniEventType type;
+  final DateTime start;
+  final Period period;
```

Comment:
Keep the name `duration` even if the type changed. It's not immediately clear what the "period" of an event represents, "duration" makes more sense.